python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -737,36 +771,45 @@ cdef class Column:
         """
         ndim = len(shape)
         flat_size = functools.reduce(operator.mul, shape)
-        data_col = Column(
-            data_type=dtype,
-            size=flat_size,
-            data=data,
-            mask=None,
-            null_count=0,
-            offset=0,
-            children=[],
-        )
 
-        int32_dtype = DataType(type_id.INT32)
+        if base is None:
+            base = Column(
+                data_type=dtype,
+                size=flat_size,
+                data=data,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[],
+            )
+
+        int32_dtype = <DataType>DataType(type_id.INT32)
+        nested = base
 
         for i in range(ndim - 1, 0, -1):
-            total_rows = functools.reduce(operator.mul, shape[:i])
-            offsets_col = sequence(
-                total_rows + 1,
+            outer_len = functools.reduce(operator.mul, shape[:i])
+
+            range_col = <Column>sequence(
+                outer_len + 1,
                 Scalar.from_py(0, int32_dtype),
-                Scalar.from_py(shape[i], int32_dtype),
+                Scalar.from_py(1, int32_dtype),
+            )
+            stride = <Scalar>Scalar.from_py(shape[i], int32_dtype)
+            offsets_col = binary_operation(
+                range_col, stride, binary_operator.MUL, int32_dtype
```

Comment:
Not sure if I'm missing something but why isn't `offsets_col = sequence(outer_len + 1, Scalar.from_py(0, int32_dtype), Scalar.from_py(shape[i], int32_dtype))` sufficient?

Reply:
It is, done in https://github.com/rapidsai/cudf/pull/18916/commits/cab6401f1f150bf48a067d570ac587aa56bac828